homeassistant/components/flipr/sensor.py

```diff
@@ -0,0 +1,85 @@
+"""Sensor platform for the Flipr's pool_sensor."""
+from homeassistant.const import (
+    ATTR_ATTRIBUTION,
+    DEVICE_CLASS_TEMPERATURE,
+    DEVICE_CLASS_TIMESTAMP,
+    TEMP_CELSIUS,
+)
+from homeassistant.helpers.entity import Entity
+
+from . import FliprEntity
+from .const import ATTRIBUTION, CONF_FLIPR_ID, DOMAIN
+
+SENSORS = {
+    "chlorine": {
+        "unit": "mV",
+        "icon": "mdi:pool",
+        "name": "Chlorine",
+        "device_class": None,
+    },
+    "ph": {"unit": None, "icon": "mdi:pool", "name": "pH", "device_class": None},
+    "temperature": {
+        "unit": TEMP_CELSIUS,
+        "icon": None,
+        "name": "Water Temp",
+        "device_class": DEVICE_CLASS_TEMPERATURE,
+    },
+    "date_time": {
+        "unit": None,
+        "icon": None,
+        "name": "Last Measured",
+        "device_class": DEVICE_CLASS_TIMESTAMP,
+    },
+    "red_ox": {
+        "unit": "mV",
+        "icon": "mdi:pool",
+        "name": "Red OX",
+        "device_class": None,
+    },
+}
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Defer sensor setup to the shared sensor module."""
+    flipr_id = config_entry.data[CONF_FLIPR_ID]
+    coordinator = hass.data[DOMAIN][config_entry.entry_id]
+
+    sensors_list = []
+    for sensor in SENSORS:
+        sensors_list.append(FliprSensor(coordinator, flipr_id, sensor))
+
+    async_add_entities(sensors_list, True)
+
+
+class FliprSensor(FliprEntity, Entity):
+    """Sensor representing FliprSensor data."""
+
+    @property
+    def name(self):
+        """Return the name of the particular component."""
+        return f"Flipr {self.flipr_id} {SENSORS[self.info_type]['name']}"
+
+    @property
+    def state(self):
+        """State of the sensor."""
+        return self.coordinator.data[self.info_type]
```

Comment:
Just noticed, based on the tests, this can be a `datetime` object as well, right??
If so, we need to make sure it is returned as an iso valid timestamp or else it will have issues on the frontend in certain browsers (e.g., Safari). Besides, `datetime` isn't a valid state.

```suggestion
        state = self.coordinator.data[self.info_type]
        if isinstance(state, datetime):
            return state.isoformat()
        return state
```

Reply:
yes, that's right, it can be a datetime.